zerver/decorator.py

```diff
@@ -183,11 +183,17 @@ def validate_api_key(request, role, api_key, is_webhook=False):
     # Remove whitespace to protect users from trivial errors.
     role, api_key = role.strip(), api_key.strip()
 
-    if is_remote_server(role):
+    if settings.ZILENCER_ENABLED and is_remote_server(role):
```

Comment:
Yes, but that state is impossible, since remote servers are part of the zilencer system.  Probably worth explaining that in the commit message, though.